### ISSUE TITLE
Move url manipulation away from core

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#3321]
Right now, some url manipulation (library importing, Web Share Target API,...) is done in the core instead of the web app.

This poses these problems:

1. Part of the logic lives in the web app, and part in the core. (https://github.com/excalidraw/excalidraw/pull/3320)
2. The component takes control of the URL (both the hash and query) which it shouldn't because then the host app is at our mercy.

### Proposal

Move the logic to host app where applicable.

### Problem

It will require the host apps to handle importing libraries (and other stuff, if we e.g. also move the web share API).

But honestly, they should be doing it anyway (and they're doing it already e.g. for the collab).

/cc @ad1992

**Comment by ad1992** [comment]
Yes I agree the package shouldn't control the url of the consumer & we should move those things into the App from core soon.
